### ISSUE TITLE
Search by field with different table name

### ISSUE THREAD

**Issue by jaulz** [LastDragon-ru/lara-asp#196]
Right now, the name of the table of is prepended to any filter defined by `@searchBy`, e.g. `users.active`. However, with custom queries I have a case where I would like to query a field on a different table of the query. I think the easiest would be to use the rename directive (e.g. ` @rename( attribute: "related_table.name" )`) and within the query enhancer check if the attribute contains a dot?

**Comment by LastDragon-ru** [comment]
> Right now, the name of the table of is prepended to any filter defined by @searchBy, e.g. users.active.

Yep (but only for Eloquent), it is required to avoid "Integrity constraint violation: 1052 Column 'xxx' in where clause is ambiguous" error when query has joins.

> I think the easiest would be to use the rename directive (e.g.  @rename( attribute: "related_table.name" )) and within the query enhancer check if the attribute contains a dot?

To be honest, I'm not sure. I always prefer explicit form (`{ relation { name }}`) and never used `@rename` :) Maybe overriding/implementation of the [`BuilderFieldResolver`](https://github.com/LastDragon-ru/lara-asp/blob/main/packages/graphql/README.md#builder-fieldcolumn-name) will be easy?

**Comment by jaulz** [comment]
Okay, thanks for the hint. I will have a look 😊
